qutebrowser/mainwindow/tabwidget.py

```diff
@@ -514,28 +514,32 @@ def tabSizeHint(self, index):
             except KeyError:
                 pinned = False
 
+            no_pinned_count = self.count() - self.pinned_count
+            pinned_width = tab_width_pinned_conf * self.pinned_count
+            no_pinned_width = self.width() - pinned_width
+
             if pinned:
-                size = QSize(tab_width_pinned_conf, height)
-                qtutils.ensure_valid(size)
-                return size
-
-            # If we *do* have enough space, tabs should occupy the whole window
-            # width. If there are pinned tabs their size will be subtracted
-            # from the total window width.
-            # During shutdown the self.count goes down,
-            # but the self.pinned_count not - this generates some odd behavior.
-            # To avoid this we compare self.count against self.pinned_count.
-            if self.pinned_count > 0 and self.count() > self.pinned_count:
-                pinned_width = tab_width_pinned_conf * self.pinned_count
-                no_pinned_width = self.width() - pinned_width
-                width = no_pinned_width / (self.count() - self.pinned_count)
+                width = tab_width_pinned_conf
             else:
-                width = self.width() / self.count()
 
-            # If width is not divisible by count, add a pixel to some tabs so
-            # that there is no ugly leftover space.
-            if index < self.width() % self.count():
+                # If we *do* have enough space, tabs should occupy the whole
+                # window # width. If there are pinned tabs their size will be
```

Comment:
Thanks for the hint, I actually found a rogue line in vimrc that set the wrong formatoptions!